src/GitHubExtension/Widgets/GitHubWidget.cs

```diff
@@ -140,7 +140,16 @@ public override void OnActionInvoked(WidgetActionInvokedArgs actionInvokedArgs)
                 break;
 
             case WidgetAction.Save:
+                // Set loading page while we swap out the data.
+                Page = WidgetPageState.Loading;
+
+                // It might take some time to get the new data, so
+                // set data state to "unknown" so that loading page is shown.
+                DataState = WidgetDataState.Unknown;
+                UpdateWidget();
+
                 SavedRepositoryUrl = string.Empty;
+                LoadContentData();

```

Comment:
```suggestion
                LoadContentData();
                
                // Reset the throttle time and force an immediate data update request.
                LastUpdated = DateTime.Min;
                RequestContentData();

```

Reply:
This should cause an immediate update by the widget.

Reply:
Updated.